usaspending_api/config/envs/default.py

```diff
@@ -75,125 +81,149 @@ def __new__(cls, *args, **kwargs):
         if not is_database_url_provided:
             # First validate that we have enough parts to provide it
             required_pg_dsn_parts = [
-                "USASPENDING_DB_HOST",
-                "USASPENDING_DB_PORT",
-                "USASPENDING_DB_NAME",
-                "USASPENDING_DB_USER",
-                "USASPENDING_DB_PASSWORD",
+                f"{db_conf_prefix}_HOST",
+                f"{db_conf_prefix}_PORT",
+                f"{db_conf_prefix}_NAME",
+                f"{db_conf_prefix}_USER",
+                f"{db_conf_prefix}_PASSWORD",
             ]
-            if not all(pg_dsn_part in values for pg_dsn_part in required_pg_dsn_parts):
-                raise ValueError(
-                    "PostgreSQL DATABASE_URL was not provided and could not be built because one or more of these "
-                    "required parts were missing. Please check that the parts are completely configured, or provide a "
-                    f"DATABASE_URL instead: {required_pg_dsn_parts}"
+            if not (
+                all(pg_dsn_part in values for pg_dsn_part in required_pg_dsn_parts)
+                and all(values[pg_dsn_part] is not None for pg_dsn_part in required_pg_dsn_parts)
+            ):
+                if required:
+                    raise ValueError(
+                        f"PostgreSQL {db_url_conf_name} was not provided and could not be built because one or more of"
+                        " these required parts were missing. Please check that the parts are completely configured, or"
+                        f" provide a {db_url_conf_name} instead: {required_pg_dsn_parts}"
+                    )
+                else:
+                    print(
+                        f"PostgreSQL {db_url_conf_name} was not provided and could not be built because one or more of"
+                        " these required parts were missing. Leaving blank for now. Please check that the parts are"
+                        f" completely configured, or provide a {db_url_conf_name} to use it:"
+                        f" {required_pg_dsn_parts}"
+                    )
+            else:
+                pg_dsn = PostgresDsn(
+                    url=None,
+                    scheme="postgres",
+                    user=values[f"{db_conf_prefix}_USER"],
+                    password=values[f"{db_conf_prefix}_PASSWORD"].get_secret_value(),
+                    host=values[f"{db_conf_prefix}_HOST"],
+                    port=values[f"{db_conf_prefix}_PORT"],
+                    path="/" + values[f"{db_conf_prefix}_NAME"] if values[f"{db_conf_prefix}_NAME"] else None,
+                )
+                values = eval_default_factory_from_root_validator(cls, values, db_url_conf_name, lambda: str(pg_dsn))
+
+        # the broker config values are less required than the usas db values. if available, check for consistency.
+        if values[db_url_conf_name]:
```

Comment:
Based on the comment you have here should this be `values.get()` instead of `values[]`?

Reply:
I can change it to the .get() equivalent. I believe usually this field is available, just defaults to `None`, but couldn't hurt to be more careful.